### PR TITLE
[REST] Apply metadata selector regex on all namespaces

### LINKED ISSUE #1692: [REST API] Metadata selector for items does not work
Followup issue for #645 since this is still not fixed

The metadata selector is supposed to be a regex, but it does not work.

Test-items:
```
String	MetaTest1	{test_ns="asdf"}
String	MetaTest2	{test_ns="qwer" [a="b", c="d"]}
```

Requesting the item with the full metadata namespace as a selector does work

```http://localhost:8080/rest/items/MetaTest1?metadata=test_ns```
```json
{
  "link": "http://localhost:8080/rest/items/MetaTest1",
  "state": "NULL",
  "metadata": {
    "test_ns": {
      "value": "asdf"
    }
  },
  "editable": false,
  "type": "String",
  "name": "MetaTest1",
  "tags": [],
  "groupNames": []
}
```

Requesting the item with a wildcard (e.g. ``.+``) as a metadata selector returns always empty metadata

```http://localhost:8080/rest/items/MetaTest1?metadata=.%2B```
```json
{
  "link": "http://localhost:8080/rest/items/MetaTest1",
  "state": "NULL",
  "editable": false,
  "type": "String",
  "name": "MetaTest1",
  "tags": [],
  "groupNames": []
}
```

### PR DESCRIPTION
**Pull request by silamon**
Fixes #1692 .

### CODE REVIEW

**Comment by spacemanspiff2007** [comment]
Who could review and merge this?
It's a small fix that would be really appreciated for OH3.

File: bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/item/MetadataSelectorMatcher.java [hunk 1]
```diff
@@ -72,19 +63,22 @@ public MetadataSelectorMatcher(final @Reference MetadataRegistry metadataRegistr
                     .map(n -> n.trim()) //
                     .collect(Collectors.toSet());
 
+            Set<String> allMetadataNamespaces = metadataRegistry.getAll().stream() //
+                    .map(metadata -> metadata.getUID().getNamespace()) //
+                    .distinct() //
+                    .collect(Collectors.toSet());
```
Comment (cweitkamp) [inline bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/item/MetadataSelectorMatcher.java:69]: Use shortcut here or remove static import.

```suggestion
                    .collect(toSet());
```

File: bundles/org.openhab.core.io.rest.core/src/test/java/org/openhab/core/io/rest/core/internal/item/MetadataSelectorMatcherTest.java [hunk 2]
```diff
@@ -53,21 +53,21 @@
 
     @BeforeEach
     public void setup() throws Exception {
-        when(configDescriptionRegistry.getConfigDescriptions(null)).thenReturn(mockConfigDescriptions());
+        when(metadataRegistry.getAll()).thenReturn(mockMetadata());
         when(metadataRegistry.isInternalNamespace(anyString())).thenReturn(false);
 
-        matcher = new MetadataSelectorMatcher(metadataRegistry, configDescriptionRegistry);
+        matcher = new MetadataSelectorMatcher(metadataRegistry);
```
Comment (cweitkamp) [inline bundles/org.openhab.core.io.rest.core/src/test/java/org/openhab/core/io/rest/core/internal/item/MetadataSelectorMatcherTest.java:59]: Private property `configDescriptionRegistry` can be removed completely from this test.

File: bundles/org.openhab.core.io.rest.core/src/test/java/org/openhab/core/io/rest/core/internal/item/MetadataSelectorMatcherTest.java [hunk 3]
```diff
@@ -53,21 +53,21 @@
 
     @BeforeEach
     public void setup() throws Exception {
-        when(configDescriptionRegistry.getConfigDescriptions(null)).thenReturn(mockConfigDescriptions());
+        when(metadataRegistry.getAll()).thenReturn(mockMetadata());
```
Comment (cweitkamp) [inline bundles/org.openhab.core.io.rest.core/src/test/java/org/openhab/core/io/rest/core/internal/item/MetadataSelectorMatcherTest.java:56]: We do not need the method `mockMetadate()`. Can be implemented like this:

```
       when(metadataRegistry.getAll()).thenReturn(List.of(new Metadata(new MetadataKey("magic", "test_item"), "test", Map.of()), ... ));
```

**Review by cweitkamp — COMMENTED** [review]
Thanks. Some minor codestyle comments.

**Review by cweitkamp — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**